src/Kmd.Logic.Gateway.Automation/PublishProducts.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Kmd.Logic.Gateway.Automation.Client;
+using Kmd.Logic.Gateway.Automation.PublishFile;
+using Kmd.Logic.Identity.Authorization;
+
+namespace Kmd.Logic.Gateway.Automation
+{
+    internal class PublishProducts
+    {
+        private readonly PublishPolicies _publishPolicies;
+        private readonly GatewayOptions options;
+        private readonly GatewayClientFactory gatewayClientFactory;
+        private List<GatewayAutomationResult> _publishResults;
+
+        public PublishProducts(HttpClient httpClient, LogicTokenProviderFactory tokenProviderFactory, GatewayOptions options, List<GatewayAutomationResult> publishResults)
+        {
+            this.options = options ?? throw new ArgumentNullException(nameof(options));
```

Comment:
if these validation are done calling method, do we required here ?

Reply:
I had warning , instead of suppressing . I added a check